koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -330,27 +330,63 @@ def _incremental_upload_logs(self, child_pid):
         finally:
             watcher.clean()
 
-    def _write_incremental_logs(self, build_id, log_filename):
-        log_basename = os.path.basename(log_filename)
+    def _write_combined_log(self, log, logs_dir):
+        log_basename = 'openshift-incremental.log'
+        log_filename = os.path.join(logs_dir, log_basename)
+
         self.logger.info("Will write follow log: %s", log_basename)
-        try:
-            build_logs = self.osbs().get_build_logs(build_id,
-                                                    follow=True)
-        except Exception, error:
-            msg = "Exception while waiting for build logs: %s" % error
-            raise ContainerError(msg)
         outfile = open(log_filename, 'w')
         try:
-            for line in build_logs:
+            for line in log:
                 outfile.write("%s\n" % line)
-                outfile.flush()
+            outfile.flush()
```

Comment:
Actually we want to flush after every line, otherwise when watching a stuck running build you don't know what's causing the delay because there are messages still buffered.